quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep

```diff
@@ -0,0 +1,119 @@
+param experimentName string
+
+@description('The base URI where artifacts required by this template are located including a trailing \'/\'.')
+param _artifactsLocation string = deployment().properties.templateLink.uri
+
+@description('The sasToken required to access _artifactsLocation.')
+@secure()
+param _artifactsLocationSasToken string = ''
+
+@description('Specifies dictionary of inputs search for sweep job.')
+param inputs object = {
+  iris_csv: {
+    mode: 'ReadOnlyMount'
+    uri: uri(_artifactsLocation, 'data/iris.csv${_artifactsLocationSasToken}')
+    jobInputType: 'uri_file'
+  }
+}
+
+@description('Specifies execution contraints for sweep job.')
+param limits object = {
+  jobLimitsType: 'Sweep'
+  timeout: 'PT20M'
+  trialTimeout: 'PT50S'
+  maxConcurrentTrials: 3
+  maxTotalTrials: 5
+}
+
+@description('Specifies objective for sweep job.')
+param objective object = {
+  goal: 'maximize'
+  primaryMetric: 'result'
+}
+
+@description('Specifies sampling algorithm for sweep job.')
+param samplingAlgorithmType string = 'Random'
+
+@description('Specifies different search space for sweep job.')
+param searchSpace object = {
+  learning_rate: [ 'uniform', [ json('0.01'), json('0.9') ] ]
+  boosting: [ 'choice', [ [ 'gbdt', 'dart' ] ] ]
+}
+
+@description('Specifies command to be executed by trials of sweep job.')
+param command string = '''python main.py --iris-csv ${{inputs.iris_csv}} --learning-rate ${{search_space.learning_rate}} --boosting ${{search_space.boosting}}'''
+
+@description('Specifies the curated environment to run sweep job.')
+param environmentName string = 'AzureML-lightgbm-3.2-ubuntu18.04-py37-cpu'
+
+// fetching existing curated environment resource object
+resource environmemt 'Microsoft.MachineLearningServices/workspaces/environments@2022-05-01' existing = {
+  name: '${workspaceName}/${environmentName}'
+}
+
+// fetching exisitng compute/ environment resource ids
+var computeId = resourceId('Microsoft.MachineLearningServices/workspaces/computes', workspaceName, computeName)
+var environmentId = resourceId('Microsoft.MachineLearningServices/workspaces/environments/versions', workspaceName, environmentName, environmemt.properties.latestVersion)
+
+// creating codeVersion resource
+module codeVersion 'modules/codeversion.bicep' = {
+  name: 'blob'
+  params: {
+    location: location
+    workspaceName: workspaceName
+    storageAccountName: storageAccountName
+  }
+}
+
+// fetching codeId from codeVersion module
+var codeId = codeVersion.outputs.codeId
+
+// creating sweep job resource
+resource sweepjobResource 'Microsoft.MachineLearningServices/workspaces/jobs@2022-06-01-preview' = {
+  name: '${workspaceName}/${jobName}'
+  properties: {
+    description: 'Sweep Job Resource from ARM Template'
+    properties: {}
+    tags: {
+      referenceNotebook: 'https://github.com/Azure/azureml-examples/blob/main/sdk/jobs/single-step/lightgbm/iris/lightgbm-iris-sweep.ipynb'
```

Comment:
what happens if this file is not found?

Reply:
It's just a tag that links to a notebook that creates a similar job as this arm template. It shows up at the AzureML Studio UI like this:

![image](https://user-images.githubusercontent.com/7882752/188832969-a17079dc-7835-4f64-9e52-cdceab397f68.png)

The link being invalid (e.g. 404) will not fail the job run.